persistence-spi/src/main/java/org/apache/unomi/persistence/spi/conditions/dispatcher/ConditionQueryBuilderDispatcherSupport.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.unomi.persistence.spi.conditions.dispatcher;
+
+import org.apache.unomi.api.conditions.Condition;
+import org.apache.unomi.persistence.spi.conditions.ConditionContextHelper;
+import org.apache.unomi.scripting.ScriptExecutor;
+import org.slf4j.Logger;
+
+import java.util.Map;
+import java.util.function.Predicate;
+
+/**
+ * Shared helper for condition query builder dispatchers (ES/OS). Centralizes logic that is
+ * backend-agnostic: contextualization, legacy ID mapping with logging, and queryBuilder key resolution.
+ * The legacy-to-new queryBuilder identifiers are centralized here in
+ * {@link #LEGACY_TO_NEW_QUERY_BUILDER_IDS} and are used by the no-arg
+ * {@link #resolveLegacyQueryBuilderId(String, String, org.slf4j.Logger)} and
+ * {@link #findQueryBuilderKey(String, String, java.util.function.Predicate, org.slf4j.Logger)} methods.
+ * This helper intentionally avoids any dependency on backend-specific query types.
+ */
+public class ConditionQueryBuilderDispatcherSupport {
```

Comment:
Instead of adding the ConditionQueryBuilderDispatcherSupport, I would suggest to create an abstract class ConditionQueryBuilderDispatcher which will contains findQueryBuilderKey
and an abstract method getLogger()
The ConditionESQueryBuilderDispatcher and ConditionOSQueryBuilderDispatcher could extends ConditionQueryBuilderDispatcher.